internal-packages/simple-cached-fetch/src/index.ts

```diff
@@ -0,0 +1,66 @@
+import { buildFetch } from '@data-eden/network';
+import { buildCache } from '@data-eden/cache';
+
+async function loggerMiddleware(
+  request: Request,
+  next: (request: Request) => Promise<Response>
+): Promise<Response> {
+  console.log('request happening!');
+  return next(request);
+}
+
+function getUrl(input: RequestInfo | URL): string {
+  if (typeof input === 'string') {
+    return input;
+  }
+
+  if ('url' in input) {
+    return input.url;
+  }
+
+  return input.toString();
+}
+
+type Signal = object;
+function buildSignal(key: string, value: object): Signal {
+  // TODO: wtf is this?
+  return value;
+}
+
+export function buildCachedFetch() {
+  const fetch = buildFetch([loggerMiddleware]);
+  const cache = buildCache();
+
+  // TODO: make this not "leak"
+  let signals = new Map<string, Signal>();
+
+  return async function (
+    input: RequestInfo | URL,
+    init?: RequestInit | undefined
+  ) {
+    const key = getUrl(input);
+
+    // assume "fetch first" (could have an option for "cache first", but that is orthogonal to testing reactivity)
+    const response = await fetch(input, init);
+    const json = (await response.json()) as object;
+
+    const tx = await cache.beginTransaction();
```

Comment:
What is `tx`? Can we have a more self-documenting variable name?

Reply:
It is a cache transaction

---

We are still working on getting API doc generation setup #15

Reply:
I was less worried about doc generation and more just thinking a descriptive variable name would make the code easier to read and follow.